sharktank/sharktank/layers/causal_llm.py

```diff
@@ -142,6 +142,56 @@ def attention_mask(
         ).to(dtype)
         return numeric_mask.to(self.device)
 
+    def chunked_attention_mask(self, attention_mask: torch.Tensor) -> torch.Tensor:
+        """Apply a chunked attention mask onto a mask."""
+        batch_seq_len = attention_mask.shape[2]
+        # TODO: handle decode step
+        start_index = 0
+        end_index = batch_seq_len
+        chunked_boolean_attention_mask = self.create_boolean_chunked_attention_mask(
+            attention_chunk_size=self.config.attention_chunk_size,
+            # TODO: handle decode step
+            start_index=start_index,
+            end_index=end_index,
+        )
+
+        return torch.where(
```

Comment:
This won't work for sharded models. is the plan to come back and extend to sharded afterwards?

Reply:
I say we worry about sharding later.